src/hooks/use-archive-error-messages.ts

```diff
@@ -20,6 +20,7 @@ export function useArchiveErrorMessages() {
 				rest_cannot_view: __(
 					"There's been an authentication error. Please log in again before sharing a site."
 				),
+				rest_site_creation_blocked: __( 'Demo sites are not available for your account.' ),
```

Comment:
Not directly related to this PR but I thought it would be good to add this error in the list of the specific errors that we have